components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/triggerapi_test.go

```diff
@@ -489,3 +490,202 @@ func createDummyServer() (*httptest.Server, string, string) {
 	port := address[colonIndex+1:]
 	return server, ip, port
 }
+
+func TestNilResp(t *testing.T) {
+	checkType := "test_check"
+
+	// Test case 1: Only AUTOMATE and CHEF_INFRA_SERVER
+	expected1 := []models.CheckTriggerResponse{
+		{
+			NodeType:  constants.AUTOMATE,
+			CheckType: checkType,
+			Result: models.ApiResult{
+				Passed:  false,
+				Skipped: true,
+				Check:   checkType,
+			},
+			Host: constants.UNKNONHOST,
+		},
+		{
+			NodeType:  constants.CHEF_INFRA_SERVER,
+			CheckType: checkType,
+			Result: models.ApiResult{
+				Passed:  false,
+				Skipped: true,
+				Check:   checkType,
+			},
+			Host: constants.UNKNONHOST,
+		},
+	}
+
+	result1 := NilResp(checkType, false, false, false)
```

Comment:
Can we have different test cases for different checks? Why are we appending it like this?